main/index.ts

```diff
@@ -288,23 +289,76 @@ ipcMain.on('unsetCurrentView', async (e) => {
   dapps.unsetCurrentView(win.frameId as string)
 })
 
-ipcMain.on('*:addFrame', (e, id) => {
-  const existingFrame = store('main.frames', id)
+let currentWorkspaceId = 1
+const getNextWorkspaceId = () => `workspace:${currentWorkspaceId++}`
 
-  if (existingFrame) {
-    windows.refocusFrame(id)
+let currentViewId = 1
+const getNextViewId = () => `view:${currentViewId++}`
+
+const createWorkspace = (initialNav: Nav) => {
+  const id = getNextWorkspaceId()
+  initialNav.views.map((view: View) => {
+    view.id = getNextViewId()
+    return view
+  })
+  store.addWorkspace({
+    id,
+    nav: [initialNav],
+    fullscreen: false
+  })
+}
+
+const navWorkspace = (id: string, space: string, data: any = {}, views = []) => {
```

Comment:
all this workspace stuff could probably benefit from being put into its own module at some point

Reply:
Once the store actions are ironed out the interface should be more minimal and generalized for all windows